### PR TITLE
feat: Add CollectionIsEmptyProcessor (SonarSource rule 1155)

### PR DESCRIPTION
**Pull request by fermadeiral**
This PR adds a processor for the rule [Collection.isEmpty() should be used to test for emptiness](https://rules.sonarsource.com/java/RSPEC-1155).

Example repair:

```diff
- if (myCollection.size() == 0) {  // Noncompliant
+ if (myCollection.isEmpty()) {
...  
- if (myCollection.size() != 0) {  // Noncompliant
+ if (!myCollection.isEmpty()) {
```

### CODE REVIEW

**Comment by fermadeiral** [comment]
The method `not(CtExpression<T> expr)` is a copy from another processor. We should have something like `SpoonUtils` to keep these utility methods together and accessible to all processors. I believe this should be handled in a different PR though.

File: src/main/java/sorald/processor/CollectionIsEmptyProcessor.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+@ProcessorAnnotation(
+        key = 1155,
+        description = "Collection.isEmpty() should be used to test for emptiness")
+public class CollectionIsEmptyProcessor extends SoraldAbstractProcessor<CtBinaryOperator<?>> {
+
+    @Override
+    protected void repairInternal(CtBinaryOperator<?> element) {
+        CtExpression<?> methodCallTarget =
+                ((CtInvocation) element.getLeftHandOperand()).getTarget();
```
Comment (slarse) [inline src/main/java/sorald/processor/CollectionIsEmptyProcessor.java:15]:
```suggestion
                ((CtInvocation<?>) element.getLeftHandOperand()).getTarget();
```

**Review by slarse — CHANGES_REQUESTED** [review]
A raw type cast, otherwise LGTM!

File: src/main/java/sorald/processor/CollectionIsEmptyProcessor.java (hunk 1 above)
Comment (fermadeiral) [reply src/main/java/sorald/processor/CollectionIsEmptyProcessor.java]: Done!

Comment (slarse) [reply src/main/java/sorald/processor/CollectionIsEmptyProcessor.java]: Nice! Just waiting for CI then I'll merge.

**Review by slarse — APPROVED** [review]

**Pull request merged**